src/client/modules/my/resultsFetcher/resultsFetcher.ts

```diff
@@ -43,8 +44,86 @@ export function getRawResults(): Promise<GameResultsData[]> {
     return new Promise((resolve) => resolve(cachedData));
 }
 
-export function extractPlayerStats(resultsData: GameResultsData[]): PlayerStatsAllGames[] {
+
+export function extractAllPlayerStats(resultsData: GameResultsData[]): PlayerStatsAllGames[] {
+    let PlayerStats: PlayerStatsAllGames[] = [];
+    calculatePlayerStats(PlayerStats,resultsData);
+   return PlayerStats;
+}
+
+export function extractTwoPlayerStats (resultsData: GameResultsData[]) {
+    let PlayerStats: PlayerStatsAllGames[] = [];
+    let gameSize: PlayersPerGame[] = [];
+    let rawDataFiltered: GameResultsData[] = [];
+    gameSize = getGameSizes(resultsData);
+    let filteredGameSize = this.gameSize.filter(game => game.player_num === 2);
```

Comment:
We could collapse 2-6 player stats functions back together where the number of players is a parameter. It was just All vs N-players that I wanted you to split.

I really like how these all just do a filter before calculatePlayerStats; it makes it really simple and composable; just change the data-set to collect stats over, and get that same collection of stats.